packages/contracts/test/core/acl/integration-tests/shared-component.ts

```diff
@@ -0,0 +1,228 @@
+    dao1 = await DAO.deploy();
+    dao2 = await DAO.deploy();
+    await managingDAO.initialize(
+      '0x',
+      ownerAddress,
+      ethers.constants.AddressZero
+    );
+    await dao1.initialize('0x', ownerAddress, ethers.constants.AddressZero);
+    await dao2.initialize('0x', ownerAddress, ethers.constants.AddressZero);
+
+    // Deploy the `TestSharedComponent`
+    const TestSharedComponent = await ethers.getContractFactory(
+      'TestSharedComponent'
+    );
+    testComponent = await TestSharedComponent.deploy();
+    await testComponent.initialize(managingDAO.address);
+
+    expectedACLAuthError = customError(
+      'ACLAuth',
+      testComponent.address,
+      testComponent.address,
+      ownerAddress,
+      ID_GATED_ACTION_ROLE
+    );
+  });
+
+  it('increments IDs', async () => {
+    expect(
+      await testComponent.callStatic.createNewObject(dao1.address)
+    ).to.be.equal(0);
+
+    const tx = await testComponent.createNewObject(dao1.address);
+    await tx.wait();
+    await ethers.provider.send('evm_mine', []);
+
+    expect(
+      await testComponent.callStatic.createNewObject(dao1.address)
+    ).to.be.equal(1);
+  });
+
+  describe('idGatedAction:', async () => {
+    let oracle: TestIdGatingOracle;
+
+    beforeEach(async () => {});
+    it('executes if the ID is allowed', async () => {
+      const allowedId = 0;
+
+      // Deploy `TestIdGatingOracle` and set the allowed ID in the constructor
+      const Oracle = await ethers.getContractFactory('TestIdGatingOracle');
+      oracle = await Oracle.deploy(allowedId);
+
+      // Grants signers[0] the permission to do ID gated actions with the deployed `TestIdGatingOracle` oracle
+      dao1.grantWithOracle(
+        testComponent.address,
+        ownerAddress,
+        ID_GATED_ACTION_ROLE,
+        oracle.address
+      );
+
+      // Deploy a new object in the `TestComponent` which will have the ID 0
+      const tx = await testComponent.createNewObject(dao1.address);
+      await tx.wait();
+      await ethers.provider.send('evm_mine', []);
+
+      // Check that the ID gated action can be executed
+      expect(testComponent.callStatic.idGatedAction(allowedId));
+    });
+
+    it('reverts if the ID does not exist', async () => {
+      const allowedId = 0;
+      const nonExistingId = 1;
+
+      // Deploy the oracle and set the allowed ID
+      const Oracle = await ethers.getContractFactory('TestIdGatingOracle');
+      oracle = await Oracle.deploy(allowedId);
+
+      // Grants signers[0] the permission to do ID gated actions with the deployed `TestIdGatingOracle` oracle
+      dao1.grantWithOracle(
+        testComponent.address,
+        ownerAddress,
+        ID_GATED_ACTION_ROLE,
+        oracle.address
+      );
+
+      // The call fails if the ID differs
+      await expect(
+        testComponent.callStatic.idGatedAction(nonExistingId)
+      ).to.be.revertedWith(customError('ObjectIdNotAssigned', nonExistingId));
```

Comment:
As the initial comment said, could we please also test that using `allowedId` does work? (adapting the proposed assertion code snippet)


```suggestion
      await expect(
        testComponent.callStatic.idGatedAction(nonExistingId)
      ).to.be.revertedWith(customError('ObjectIdNotAssigned', nonExistingId));
      
      await expect(
        testComponent.callStatic.idGatedAction(allowedId)
      ).to.not.be.reverted();
```

Reply:
Fixed